### ISSUE TITLE
collector_test.go: failing regress test

### ISSUE THREAD

**Issue by bassosimone** [ooni/probe-engine#197]
See https://travis-ci.org/ooni/probe-engine/builds/630632400#L619:

```
goroutine 37 [running]:
net/http.(*conn).serve.func1(0xc00024e5a0)
	/home/travis/.gimme/versions/go1.13.5.linux.amd64/src/net/http/server.go:1767 +0x139
panic(0xc714c0, 0xc0002bb510)
	/home/travis/.gimme/versions/go1.13.5.linux.amd64/src/runtime/panic.go:679 +0x1b2
github.com/ooni/probe-engine/collector_test.TestEndToEnd.func1(0xef81e0, 0xc00027d960, 0xc000101900)
	/home/travis/gopath/src/github.com/ooni/probe-engine/collector/collector_test.go:194 +0x314
net/http.HandlerFunc.ServeHTTP(0xe00db8, 0xef81e0, 0xc00027d960, 0xc000101900)
	/home/travis/.gimme/versions/go1.13.5.linux.amd64/src/net/http/server.go:2007 +0x44
net/http.serverHandler.ServeHTTP(0xc00027d5e0, 0xef81e0, 0xc00027d960, 0xc000101900)
	/home/travis/.gimme/versions/go1.13.5.linux.amd64/src/net/http/server.go:2802 +0xa4
net/http.(*conn).serve(0xc00024e5a0, 0xefa7e0, 0xc0001a1500)
	/home/travis/.gimme/versions/go1.13.5.linux.amd64/src/net/http/server.go:1890 +0x875
created by net/http.(*Server).Serve
	/home/travis/.gimme/versions/go1.13.5.linux.amd64/src/net/http/server.go:2928 +0x384
--- FAIL: TestEndToEnd (0.00s)
    collector_test.go:228: Post http://127.0.0.1:34093/report/_id: EOF
```

**Comment by bassosimone** [comment]
This does not happen with master, tho, only with #196. So maybe it's caused by #196?

**Comment by bassosimone** [comment]
Fixed in d2dc6e0. It was indeed a #196-only thing.
